File: mns-reactor/src/main/java/cz/fb/manaus/reactor/betting/listener/MarketSnapshotListener.java
package cz.fb.manaus.reactor.betting.listener;

import cz.fb.manaus.core.model.AccountMoney;
import cz.fb.manaus.core.model.MarketSnapshot;
import cz.fb.manaus.reactor.betting.BetCollector;

import java.util.Optional;
import java.util.Set;

public interface MarketSnapshotListener {

    void onMarketSnapshot(MarketSnapshot snapshot,
                          BetCollector betCollector,
                          Optional<AccountMoney> accountMoney,
                          Set<String> categoryBlackList);

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/betting/listener/MarketSnapshotListener.kt
package cz.fb.manaus.reactor.betting.listener

import cz.fb.manaus.core.model.MarketSnapshotEvent
import cz.fb.manaus.reactor.betting.BetCommand

interface MarketSnapshotListener {

    fun onMarketSnapshot(marketSnapshotEvent: MarketSnapshotEvent): List<BetCommand>

}
